src/bug-filing/common/http-query-builder.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { size, values } from 'lodash';
+import { DictionaryStringTo } from '../../types/common-types';
+
+export type QueryParam = {
+    name: string;
+    value: string;
+};
+
+export class HTTPQueryBuilder {
+    public static readonly maxUrlLength = 3000;
+    private baseUrl: string;
+    private parameters: DictionaryStringTo<QueryParam>;
+
+    public withBaseUrl(baseUrl: string): HTTPQueryBuilder {
+        this.baseUrl = baseUrl;
+        return this;
+    }
+
+    public withParam(name: string, value: string): HTTPQueryBuilder {
+        if (this.parameters == null) {
+            this.parameters = {};
+        }
+
+        this.parameters[name] = {
+            name,
+            value,
+        };
+
+        return this;
+    }
+
+    public build(): string {
+        if (size(this.parameters) === 0) {
+            return this.baseUrl;
+        }
+
+        const queryParameters = values(this.parameters)
+            .map(param => `${encodeURIComponent(param.name)}=${encodeURIComponent(param.value)}`)
+            .join('&');
+
+        const fullUrl = `${this.baseUrl}?${queryParameters}`;
+
+        return this.truncate(fullUrl);
+    }
+
+    private truncate(url: string): string {
+        if (url.length <= HTTPQueryBuilder.maxUrlLength) {
+            return url;
+        }
+
+        const truncateIndex = url.substr(0, HTTPQueryBuilder.maxUrlLength).length;
```

Comment:
Won't this always return HTTPQueryBuilder.maxUrlLength? I'm confused.

Reply:
[Original code](https://dev.azure.com/mseng/1ES/_git/accessibility-insights-web/commit/7d3c4e5e17ebf93535c22691732ebe73fbfe9efa?path=%2Fsrc%2Fbackground%2Fbug-url-builder.ts&gridItemType=2&_a=contents
) had a little extra logic here to truncate before a %3C (<) so we didn't cut things off in the middle of a parameter name, HTML element, or escape sequence. We probably still want to do something similar here.

Reply:
It didn't occur to me at first reading the code. I remove it while writing the tests for it but now I see why we need it. So I added it back